core/iwasm/interpreter/wasm_loader.c

```diff
@@ -2681,6 +2851,96 @@ load_datacount_section(const uint8 *buf, const uint8 *buf_end,
 }
 #endif
 
+#if WASM_ENABLE_TAGS != 0
+static bool
+load_tag_section(const uint8 *buf, const uint8 *buf_end, const uint8 *buf_code,
+                 const uint8 *buf_code_end, WASMModule *module, char *error_buf,
+                 uint32 error_buf_size)
+{
+    LOG_VERBOSE("In %s\n", __FUNCTION__);
+    const uint8 *p = buf, *p_end = buf_end;
+    size_t total_size = 0;
+    uint32 section_tag_count = 0; // number of tags defined in the section
+    uint8 tag_attribute;
+    uint32 tag_type;
+
+    _EXCEVERBOSE("%s buf=%p buf_end=%p, len=%ld\n", __FUNCTION__, buf, buf_end,
+                 (long)buf_end - (long)buf);
+
+    /* get tag count */
+    read_leb_uint32(p, p_end, section_tag_count);
+    module->tag_count =
+        module->import_tag_count
+        + section_tag_count; // total tags (imported and section)
+    _EXCEVERBOSE("%s: import_tag_count = %d, section_tag_count = %d\n",
+                 __FUNCTION__, module->import_tag_count, section_tag_count);
+
+    if (section_tag_count) {
+        total_size = sizeof(WASMTag) * module->tag_count;
+        if (!(module->tags =
+                  loader_malloc(total_size, error_buf, error_buf_size))) {
+            return false;
+        }
+        /* load each tag, imported tags precede the tags */
+        for (uint32 tag_index = module->import_tag_count;
```

Comment:
Had better declare `uint32 tag_index` above.

Reply:
Updated code according to coding style